reconcile/openshift_routes.py

```diff
@@ -28,3 +30,17 @@ def run(
         cluster_name=cluster_name,
         namespace_name=namespace_name,
     )
+
+
+def early_exit_desired_state(*args, **kwargs) -> dict[str, Any]:
+    namespaces, _ = orb.get_namespaces(PROVIDERS)
+    resources = [
+        orb.fetch_provider_resource(r["path"]).body
```

Comment:
shouldn't you filter on the `route` provider here?

Reply:
it's filtered in `orb.get_namespaces`.

Reply:
oh i just realized `canonicalize_namespace` is doing in-place filtering based on the providers. look good them